### PR TITLE
Upgrading redis to v6 + improvements on setup

### PR DESCRIPTION
**Pull request by piggito**
Improvements on redis setup:

1. Upgrade docker image to v6 cause v5 stopped at 5.0.9 which is EOL since 2019 https://docs.redislabs.com/latest/rs/administering/product-lifecycle/

2. Fix issues in redis report `/admin/reports/redis`

3. Include `make redis-settings` goal and `settings.redis.php` file to simplify basic setup

### CODE REVIEW

File: docker/redis/redis.conf [hunk 1]
```diff
@@ -0,0 +1,1812 @@
+# Redis configuration file example.
```
Comment (davidferlay) [inline docker/redis/redis.conf:1]: Is that file required ? I don't recall we ever used it on projects till now

**Comment by andypost** [comment]
Moreover there's no support for redis 6 in PHP extension, first RC is out today https://pecl.php.net/package-changelog.php?package=redis&amp;release=5.3.0RC1

File: docker/redis/redis.conf (hunk 1 above)
Comment (andypost) [reply docker/redis/redis.conf:1]: not needed, should be removed in favour of `command` addition

Comment (davidferlay) [reply docker/redis/redis.conf]: Removed commented and empty lines in docker/redis/redis.conf to make it clearer what to move

File: Makefile [hunk 2]
```diff
@@ -92,8 +92,15 @@ ifneq ($(strip $(MODULES)),)
...
-
 endif
+	# Uncomment for redis support.
+	# make -s redis-settings
+
+redis-settings:
+	@echo "Turn on settings.redis"
+	$(call php, chmod +w web/sites/default)
+	$(call php, cp settings/settings.redis.php web/sites/default/settings.redis.php)
+	$(call php-0, sed -i "/settings.redis.php';/s/# //g" web/sites/default/settings.php)
```
[the remark this answers is not in the document]
Comment (andypost) [reply Makefile:103]: it replaces this line with other settings.php file https://git.drupalcode.org/project/drupal/-/blob/9.0.x/sites/default/default.settings.php#L774
